R/space.R

```diff
@@ -109,3 +109,77 @@ space_cjk <- function(text) {
     class = "unicode_block_name_error"
   )
 }
+
+#' Add Spaces Around Punctuation
+#'
+#' To keep punctuation during tokenization, it's convenient to add spacing
+#' around punctuation. This function does that, with options to keep certain
+#' types of punctuation together as part of the word.
+#'
+#' @inheritParams validate_utf8
+#' @param hyphens Logical; should hyphens be treated as punctuation?
+#' @param abbreviations Logical; should ' between letters be treated as
+#'   punctuation?
+#'
+#' @return A character vector the same length as the input text, with spaces
+#'   added around punctuation characters.
+#' @export
+#'
+#' @examples
+#' to_space <- "This is some 'gosh-darn' $5 text. Isn't it lovely?"
+#' to_space
+#' space_punctuation(to_space)
+#' space_punctuation(to_space, hyphens = FALSE)
+#' space_punctuation(to_space, abbreviations = FALSE)
+space_punctuation <- function(text, hyphens = TRUE, abbreviations = TRUE) {
+  # This feels hacky but I can't find a better way than to protect the things I
+  # want to preserve.
+  if (!hyphens) {
+    text <- stringr::str_replace_all(
+      text,
+      "-",
```

Comment:
The problem is I'd want to also preserve it for something like "some- -thing", which regexly looks very much like "some--thing". BUT I think I can make sure it touches a letter and either whitespace and another letter...
A current problem: \w also catches numerals, so 4-2 would remain 4-2 rather than 4 - 2. But if I use [A-Za-z] I miss accented characters and such, so I think I want to leave it for now and see what C4 shows us; it might be a case where we want to convert numerals to words before everything else in the long run, I'm not sure.